xocto/ranges.py

```diff
@@ -886,6 +886,24 @@ def seconds(self) -> int:
         """
         return int((self.end - self.start).total_seconds())
 
+    def localize(self, tz: datetime.tzinfo) -> FiniteDatetimeRange:
+        """
+        Returns the range with boundaries adjusted to the specified timezone.
+
+        See datetime.astimezone for more details.
+
+        Raises:
+            ValueError:
+                If one or both boundaries are naive (no timezone).
+        """
+        if not self.start.tzinfo or not self.end.tzinfo:
```

Comment:
🧀 `astimezone` will assume the system timezone by default [^1], which may or may not be correct - best to error, which is consistent with the behaviour for `localtime` / Django.

[^1]: https://docs.python.org/3/library/datetime.html#datetime.datetime.astimezone